### ISSUE TITLE
Refiner erases details

### ISSUE THREAD

**Issue by Alexadar** [huggingface/diffusers#4921]
### Describe the bug

I'm experimenting with refiner and fine-tuned models, and I'm discovered that refiner erases details.

### Reproduction

1) download https://civitai.com/models/122822/crystal-clear-xl?modelVersionId=133832
2) run in notebook

```
from diffusers import DiffusionPipeline, StableDiffusionXLPipeline, DPMSolverMultistepScheduler
import torch

base = StableDiffusionXLPipeline.from_single_file(
    "/home/oleksandr/projects/github/ComfyUI/models/checkpoints/crystalClearXL_ccxl.safetensors", 
    torch_dtype=torch.float16
)
base.scheduler = DPMSolverMultistepScheduler.from_config(base.scheduler.config, use_karras_sigmas=True)
base.to("cuda")

refiner = DiffusionPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-refiner-1.0",
    text_encoder_2=base.text_encoder_2,
    vae=base.vae,
    torch_dtype=torch.float16,
    use_safetensors=True,
    variant="fp16",
)
refiner.scheduler = DPMSolverMultistepScheduler.from_config(refiner.scheduler.config, use_karras_sigmas=True)
refiner.to("cuda")

prompt = "top down view of trees in forest"
image = base(
    prompt=prompt,
    num_inference_steps=40,
    generator = torch.Generator(device="cuda").manual_seed(471517296243298)
).images
image[0].save('base.png')
image = refiner(
    prompt=prompt,
    num_inference_steps=40,
    image=image,
    generator = torch.Generator(device="cuda").manual_seed(383071704850877)
).images
image[0].save('refined.png)
```
<img width="1018" alt="image" src="https://github.com/huggingface/diffusers/assets/14125937/6fc47f6a-4806-4519-97a0-2d7b0fb71f0b">
<img width="1016" alt="image" src="https://github.com/huggingface/diffusers/assets/14125937/fe35ed83-69df-4031-9966-7f4c6e216a8b">

As you can see from uploads refiner reduced details, is it possible to fix it somehow? I tried the same in ComfyUI and it seems that refiner works better, it's not removing details

### System Info

diffusers-0.20.2

**Comment by sayakpaul** [comment]
> I tried the same in ComfyUI and it seems that refiner works better, it's not removing details

Did you use the exact same inference-time parameters in ComfyUI? I would appreciate it if you could double check ALL the hyperparameter values on your end and ensure that the basis for comparison is consistent.

**Comment by Alexadar** [comment]
here is comfyui workflow and screenshots.
https://gist.github.com/Alexadar/d42617617a3f92b19a540343f8b68672
Images are better , not sure why, but main point that refiner improve image, rather than distort

<img width="747" alt="image" src="https://github.com/huggingface/diffusers/assets/14125937/4b7f6842-3a5f-4db0-b8ef-853fc1059194">
<img width="552" alt="image" src="https://github.com/huggingface/diffusers/assets/14125937/ea0f10c8-5e1a-43bb-a86f-a0fea3188548">

**Comment by sayakpaul** [comment]
@Alexadar I am not very familiar with the workflow file. I'd appreciate it if you could help me find out the values of the following things:

* Guidance scale
* Number of inference steps 
* Seed 
* Schedulers and their configurations 

Additionally, I welcome you to check out this section of the docs that shows how to use the base and the refiner models for a kind of "mixture of expert denoising": https://huggingface.co/docs/diffusers/main/en/using-diffusers/sdxl#base-refiner-model. You might get better results with this approach.

**Comment by Alexadar** [comment]
<img width="1331" alt="image" src="https://github.com/huggingface/diffusers/assets/14125937/195b2409-aa07-4279-9536-0a9d4c3af247">
@sayakpaul 
Guidance scale: 8
Number of inference steps: 40 base, 10 refiner
Seed: 471517296243298 base, 383071704850877 refiner
Schedulers and their configurations
dpmpp_sde, karras sigmas, add_noise true (not sure what it is)

**Comment by sayakpaul** [comment]
I think you're not using those exact hyperparameters here: https://github.com/huggingface/diffusers/issues/4921#issue-1883980551

**Comment by Alexadar** [comment]
> I think you're not using those exact hyperparameters here: [#4921 (comment)](https://github.com/huggingface/diffusers/issues/4921#issue-1883980551)

Updated generators. steps are the same in inference for refiner. Which else parameters may differ?

**Comment by NullSenseStudio** [comment]
In your example you aren't passing latents directly to the refiner like in the comfyui workflow. I think the loss in detail is due to those latents being re-encoded. You can get the latents from your base pipeline with argument `output_type="latent"`.

**Comment by Alexadar** [comment]
It works the same with “latents” and different samplers
